presto-mongodb/src/test/java/io/prestosql/plugin/mongodb/BaseMongoDistributedQueries.java

```diff
@@ -13,33 +13,21 @@
  */
 package io.prestosql.plugin.mongodb;
 
-import com.google.common.collect.ImmutableMap;
 import io.prestosql.testing.AbstractTestDistributedQueries;
-import io.prestosql.testing.QueryRunner;
 import io.prestosql.testing.sql.TestTable;
-import io.prestosql.tpch.TpchTable;
 import org.testng.SkipException;
 import org.testng.annotations.AfterClass;
 import org.testng.annotations.Test;
 
 import java.util.Optional;
 
-import static io.prestosql.plugin.mongodb.MongoQueryRunner.createMongoQueryRunner;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 @Test
-public class TestMongoDistributedQueries
+public abstract class BaseMongoDistributedQueries
         extends AbstractTestDistributedQueries
 {
-    private MongoServer server;
-
-    @Override
-    protected QueryRunner createQueryRunner()
```

Comment:
Maybe leave this method and instead provide `createMongoServer`?

Another option (preferable by me) would be to remove `createQueryRunner` method and `server` field, and maintain it in the class that creates it. More code is duplicated, but I find inheritance of infrastructure as bad practice.